### DISCUSSION TITLE
Cannot use Set to increment column value

### DISCUSSION THREAD

**Discussion by danxcraig** [uptrace/bun#625] [Q&A] [answered]
This issue was encountered while attempting to use `Set` as part of an [`UPDATE`](https://bun.uptrace.dev/guide/query-update.html#api) to a PostgreSQL database table.

In order to increment the value of an INT or BIGINT in a column in PostgreSQL, I can use:

```
UPDATE some_table
   SET some_column = some_column + 1
WHERE some_other_column = 'change me';
```
The challenge is that Bun's `Set` always wraps single quotes around it's second parameter. Thus using `Set("some_column = ?", "some_column + 1")` in code becomes `SET some_column = 'some_column + 1'` in the UPDATE statement, which subsequently fails with an _`invalid input syntax for type`_ error where the column is INT or BIGINT.

I'm assuming I'm missing something, so I'd appreciate pointers around the recommended approach for incrementing or decreasing column int values as part of an [`UPDATE`](https://bun.uptrace.dev/guide/query-update.html#api).

**Update**
As a temporary measure I'm using a separate SELECT statement to scan the existing INT from the table into a currentValue var, incrementing _that_ by one, then using that to setting the column value. It works, but it's not very elegant or efficient.

**Comment by vmihailenco** [comment, marked as answer]
You can use `q.Set("col = col + ?", 1)` or `q.Set("col = ?", bun.Safe("col + 1"))`

**Reply by danxcraig** [reply]
How did I miss that. Thank you!
